unlock-app/src/components/interface/checkout/main/Confirm.tsx

```diff
@@ -349,6 +359,53 @@ export function Confirm({
         return getReferrer(recipient, paywallConfig)
       })
 
+      const onErrorCallback = (error: Error | null, hash: string | null) => {
+        setIsConfirming(false)
+        if (error) {
+          send({
+            type: 'CONFIRM_MINT',
+            status: 'ERROR',
+            transactionHash: hash!,
+          })
+        } else {
+          if (!paywallConfig.pessimistic && hash) {
+            communication?.emitTransactionInfo({
+              hash,
+              lock: lockAddress,
+            })
+            communication?.emitUserInfo({
+              address: account,
+              signedMessage: messageToSign?.signature,
+            })
+          }
+          send({
+            type: 'CONFIRM_MINT',
+            status: paywallConfig.pessimistic ? 'PROCESSING' : 'FINISHED',
+            transactionHash: hash!,
+          })
+        }
+      }
+
+      const swap =
+        payment.method === 'swap_and_purchase'
+          ? {
+              srcTokenAddress: payment.route.quote.currency.address,
+              uniswapRouter: payment.route.swapRouter,
+              swapCallData: payment.route.swapCalldata,
+              amountInMax: ethers.utils
+                .parseUnits(
+                  payment.route
+                    .convertToQuoteToken(pricingData!.total.toString())
+                    .toFixed(4),
+                  payment.route.quote.currency.decimals
+                )
+                // 1% slippage buffer
+                .mul(101)
+                .div(100)
```

Comment:
Added this since sometimes it was undervaluing. Maybe it could be lower like 0.5%?

Reply:
I think we should optimize for the tx passing, not for the best price. Given the small amounts, even 3% is better than a failed tx (because the gas would be wasted!)

Reply:
ok, 1% is fine for now. Works for small amounts since that's what I'm testing with.